scripts/elf_helper.py

```diff
@@ -512,9 +512,7 @@ def find_kobjects(self, syms):
                 continue
 
             _, user_ram_allowed = kobjects[ko.type_obj.name]
-            if (not user_ram_allowed and
-                    (addr >= app_smem_start and addr < app_smem_end)):
-
+            if not user_ram_allowed and app_smem_start <= addr < app_smem_end:
```

Comment:
don't you need parentheses? `if not user_ram_allowed and (app_smem_start <= addr < app_smem_end)`

Reply:
Works without parentheses. `<=`/`<` have higher precedence than `and`.

Reply:
It's like

```c
if (!user_ram_allowed && x < y && y < z)
```

in C, except Python supports turning the `x < y && y < z` part into `x < y < z`. Get the same precedence.